registry/storage/driver/storagedriver.go

```diff
@@ -32,6 +32,32 @@ func (version Version) Minor() uint {
 // CurrentVersion is the current storage driver Version.
 const CurrentVersion Version = "0.1"
 
+// WalkOptions are options for Walk
+type WalkOptions struct {
+	// StopOnErrSkipDir mutates the default Walk behavior to stop whenever ErrSkipDir is returned
+	// by the callback fn, regardless if the callback fileInfo is a directory.
+	StopOnErrSkipDir bool
+}
+
+// WalkOpt is a functional setter for WalkOptions
+type WalkOpt func(*WalkOptions)
+
+// ToWalkOptions converts a slice of WalkOpt to WalkOptions
+func ToWalkOptions(opts ...WalkOpt) WalkOptions {
+	walkOpts := WalkOptions{}
+	for _, opt := range opts {
+		opt(&walkOpts)
+	}
+	return walkOpts
+}
+
+// WalkWithStopOnErrSkipDir functionally sets WalkOptions.StopOnErrSkipDir
+func WalkWithStopOnErrSkipDir(b bool) WalkOpt {
```

Comment:
just use `WithStopOnErrSkipDir` as the func name?

Reply:
I'd prefer to call this `WithStopOnErrSkipDir`, too 😅 🙇